File: SatinalmaPro.Shared/Models/TalepTurleri.cs
namespace SatinalmaPro.Shared.Models;

public static class TalepTurleri
{
    public const string Acil = "Acil";
    public const string Normal = "Normal";

    public static IReadOnlyList<string> Tum { get; } = [Acil, Normal];

    public static string GorunenAd(string tur) => tur switch
    {
        Acil => "Acil",
        Normal => "Normal",
        _ => tur
    };

    public static string TurkceAd(string tur) => tur switch
    {
        Acil => "Acil Talep",
        Normal => "Normal Talep",
        _ => tur
    };
}

File: SatinalmaPro.Android/app/src/main/java/com/satinalmapro/android/core/roles/TalepTurleri.kt
package com.satinalmapro.android.core.roles

object TalepTurleri {
    const val ACIL = "Acil"
    const val ONCELIKLI = "Oncelikli"
    const val NORMAL = "Normal"

    val TUM = listOf(NORMAL, ONCELIKLI, ACIL)

    fun fromIndex(index: Int): String = when (index.coerceIn(0, 2)) {
        2 -> ACIL
        1 -> ONCELIKLI
        else -> NORMAL
    }

    fun gorunenAd(tur: String): String = when (tur) {
        ACIL -> "Acil"
        ONCELIKLI -> "Öncelikli"
        else -> "Normal"
    }
}
